server/segment_rpc.go

```diff
@@ -434,16 +435,24 @@ func SubmitSegment(sess *BroadcastSession, seg *stream.HLSSegment, nonce uint64,
 		return nil, err
 	}
 
+	// timeout for the whole HTTP call: segment upload, transcoding, reading response
+	httpTimeout := common.HTTPTimeout
 	// set a minimum timeout to accommodate transport / processing overhead
-	dur := common.HTTPTimeout
 	paddedDur := 4.0 * seg.Duration // use a multiplier of 4 for now
-	if paddedDur > dur.Seconds() {
-		dur = time.Duration(paddedDur * float64(time.Second))
+	if paddedDur > httpTimeout.Seconds() {
+		httpTimeout = time.Duration(paddedDur * float64(time.Second))
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), dur)
+	// timeout fot the segment upload, until HTTP returns OK 200
+	uploadTimeout := time.Duration(0.5 * seg.Duration * float64(time.Second)) // use 0.5 multiplier
```

Comment:
I think this comment is unnecessary, since the code already states that we are multiplying by `0.5`. It's similar to the "multipler of 4" above which I also think could be removed IMO.
```suggestion
	uploadTimeout := time.Duration(0.5 * seg.Duration * float64(time.Second))
```

We can create `consts` for it to give them a name and make it more explicit, which I think would be better than these comments in the end. Maybe `segHttpPushTimeoutMultipler` and `segUploadTimeoutMultiplier`?

Reply:
Yes, good idea. Updated in https://github.com/livepeer/go-livepeer/pull/2122/commits/b582d2531f254390197abe680f3af39857dadcd8